### PR TITLE
Sanitise production database 

### PR DESCRIPTION
**Pull request by vigneshmsft**
### Context

Provide a sanitised database copy for debugging/local development purposes.

https://trello.com/c/15unia02/2694-sanitise-production-database

### Changes proposed in this pull request

SQL script to sanitise user, trainee data.
Amend the daily database backup workflow to sanitise 
the backup using the above script and upload sanitised dump as artifact.

### Guidance to review
Are the scripts accurate?

### CODE REVIEW

**Review by AbigailMcP — COMMENTED** [review]
Looks good! I ran it locally to check and works grand 👍 

Just a couple of comments in line, plus: do we need to think about sanitising audit logs? They're stored in the `audits` table.

File: db/scripts/sanitise.sql [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+DELETE FROM "audits";
+DELETE FROM "blazer_audits";
+DELETE FROM "validation_errors";
+DELETE FROM "apply_application_sync_requests";
+
+-- Users
+UPDATE "users"
+SET
+  dfe_sign_in_uid = null,
+  dttp_id = null,
```
Comment (defong) [inline db/scripts/sanitise.sql:11]: Can we retain the original `ids` these as it is not really needed to be sanitised.

File: db/scripts/sanitise.sql [hunk 2]
```diff
@@ -0,0 +1,33 @@
+DELETE FROM "sessions";
+DELETE FROM "audits";
+DELETE FROM "blazer_audits";
+DELETE FROM "validation_errors";
+DELETE FROM "apply_application_sync_requests";
+
+-- Users
+UPDATE "users"
+SET
+  dfe_sign_in_uid = null,
```
[the remark this answers is not in the document]
Comment (vigneshmsft) [reply db/scripts/sanitise.sql:10]: just wanted to have different first names, other suggestions have been updated.
Also all records in the `audits` is being deleted.

**Review by AbigailMcP — APPROVED** [review]

File: db/scripts/sanitise.sql (hunk 1 above)
Comment (darokel) [reply db/scripts/sanitise.sql:11]: +1

**Review by defong — CHANGES_REQUESTED** [review]
We need anonymise `first`, `middle` names as well `last` name, as they are quantifiers.

File: db/scripts/sanitise.sql [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+DELETE FROM "audits";
+DELETE FROM "blazer_audits";
+DELETE FROM "validation_errors";
+DELETE FROM "apply_application_sync_requests";
+
+-- Users
+UPDATE "users"
+SET
+  first_name = 'User',
+  last_name = concat('RegisterUser', id),
```
Comment (defong) [inline db/scripts/sanitise.sql:11]:
```suggestion
  last_name = concat('RegisterUser_', id),
```

File: db/scripts/sanitise.sql [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+WHERE email NOT LIKE '%@digital.education.gov.uk'
+      AND email NOT LIKE '%@education.gov.uk';
+
+-- Trainees
+UPDATE "trainees"
+SET
+  date_of_birth = '2000-01-01',
+  first_name = 'Trainee',
+  middle_names = null,
+  last_name = concat('TraineeUser', id),
```
Comment (defong) [inline db/scripts/sanitise.sql:22]:
```suggestion
  last_name = concat('TraineeUser_', id),
```

File: db/scripts/sanitise.sql [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  town_city = CASE WHEN town_city IS NULL THEN NULL ELSE 'London' END,
+  postcode = CASE WHEN postcode IS NULL THEN NULL ELSE 'SW1P 3BT' END,
+  international_address = CASE WHEN international_address IS NULL THEN NULL ELSE 'International Address' END;
+
+-- Dttp Users
+UPDATE "dttp_users"
+SET
+  email = concat('test_dttp_user', id, '@example.org'),
+  first_name = 'Dttp',
+  last_name = concat('DttpUser', id)
```
Comment (defong) [inline db/scripts/sanitise.sql:35]:
```suggestion
  last_name = concat('DttpUser_', id)
```

**Review by defong — APPROVED** [review]
Looks awesome!

Just optional underscoring

**Pull request merged**